extensions/shared/components/audio-player/index.jsx

```diff
@@ -0,0 +1,83 @@
+/* global _wpmejsSettings, MediaElementPlayer */
+
+/**
+ * External dependencies
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useRef } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { speak } from '@wordpress/a11y';
+
+/**
+ * Internal dependencies
+ */
+import { STATE_PLAYING, STATE_PAUSED, STATE_ERROR } from './constants';
+
+const meJsSettings = typeof _wpmejsSettings !== 'undefined' ? _wpmejsSettings : {};
+
+function AudioPlayer( {
+	trackSource,
+	handlePlay,
+	handlePause,
+	handleError,
+	playStatus = STATE_PAUSED,
+} ) {
+	const audioRef = useRef();
+
+	/**
+	 * Play current audio.
+	 *
+	 * @public
+	 */
+	const play = () => {
+		// Ignoring exceptions as they are handled globally from the audio element.
+		audioRef.current.play().catch( () => {} );
+	};
+
+	/**
+	 * Pause current audio.
+	 *
+	 * @public
+	 */
+	const pause = () => {
+		audioRef.current.pause();
+		speak( __( 'Paused', 'jetpack' ), 'assertive' );
+	};
+
+	useEffect( () => {
+		const audio = audioRef.current;
+		// Initialize MediaElement.js.
+		const mediaElement = new MediaElementPlayer( audio, meJsSettings );
+		handlePlay && audio.addEventListener( 'play', handlePlay );
+		handlePause && audio.addEventListener( 'pause', handlePause );
+		handleError && audio.addEventListener( 'error', handleError );
+
+		return () => {
+			// Cleanup.
+			mediaElement.remove();
+			handlePlay && audio.removeEventListener( 'play', handlePlay );
+			handlePause && audio.removeEventListener( 'pause', handlePause );
+			handleError && audio.removeEventListener( 'error', handleError );
+		};
+	}, [ audioRef, handlePlay, handlePause, handleError ] );
+
+	useEffect( () => {
+		const [ audioStatus, action ] =
+			audioRef.current?.paused === false ? [ STATE_PLAYING, pause ] : [ STATE_PAUSED, play ];
+		if ( STATE_ERROR !== playStatus && audioStatus !== playStatus ) {
+			action();
+		}
+	}, [ audioRef, playStatus, trackSource ] );
```

Comment:
I'm seeing the play and pause jumping back and forth when I click around the time slider / scrub bar:
![audioplayerblock](https://user-images.githubusercontent.com/1689238/100146211-8e5e8380-2e67-11eb-9744-a577982a71d1.gif)
Removing the `playStatus` argument fixes it, so maybe a change to `playStatus` is triggered when the bar is clicked. 🤔

Reply:
That's strange, I can't replicate that at the moment. It certainly looks like some of the events are fighting against each other. 

I wonder if it has got in an error state and we're trying to play the podcast? Is there anything in the console?

Reply:
> I wonder if it has got in an error state and we're trying to play the podcast? Is there anything in the console?

I'm not seeing any new console errors / warnings when it happens. [Here's a post where it's happening for me](https://happyanhinga.wpcomstaging.com/2020/11/24/youre-wrong-about-take-3/) in Firefox 84.

Reply:
That's very odd, I can't replicate this. I might have inadvertently fixed it with the other changes I've been making. Once the PR is updated, it would be great if you could retest it.

Reply:
> That's very odd, I can't replicate this. I might have inadvertently fixed it with the other changes I've been making. Once the PR is updated, it would be great if you could retest it.

I can't add back the block I was using. 🤔 Maybe I was trying to use the old audio player?
<img width="749" alt="Screen Shot 2020-12-01 at 4 30 39 PM" src="https://user-images.githubusercontent.com/1689238/100799135-9b90ea80-33f2-11eb-9555-43c366b17d2a.png">

Reply:
Maybe the block isn't building properly now? I'll check I haven't missed committing something

Reply:
can you take a look at the client console? Any error there?

Reply:
There is:
<img width="973" alt="Screen Shot 2020-12-03 at 9 06 24 AM" src="https://user-images.githubusercontent.com/1689238/101029450-0d7a4880-3547-11eb-8d5b-d3b75578c2a6.png">

Reply:
I can use that feed on my development site with this branch - the block renders correctly and I can jump around the track without seeing the original bug you can.

Everything seems to be committed and pushed, so perhaps it's something to do with the build?

Reply:
the second one seems to be a building issue

Reply:
![audio-player-test](https://user-images.githubusercontent.com/77539/101176594-7a641000-3625-11eb-9653-d0a81fc2dad5.gif)

I'm not able to reproduce it even in Firefox :-/ It works as expected to me.

Reply:
I've managed to reproduce this, but have fixed it by debouncing the events.